drivers: adc: stm32: Enable DMA based on dt configuration (#94870)
zephyrproject-rtos/zephyr: doc/releases/migration-guide-4.3.rst

  ``adcx`` for the register clock, ``adc-ker`` for the kernel source clock, and ``adc-pre`` to set
  the ADC prescaler (for series where it is located in the RCC registers).

> * :dtcompatible:`st,stm32-adc` and its derivatives now automatically enable :kconfig:option:`CONFIG_DMA`

Reviewer: ```suggestion
* STM32 ADC driver's Kconfig now automatically enables :kconfig:option:`CONFIG_DMA`
```

>   and switch driver to DMA operation mode if the ``dmas`` property is added to a node in Device Tree.

Reviewer: ```suggestion
  and switch driver to DMA operation mode if the ``dmas`` property is added to an ADC node in DeviceTree.
```
